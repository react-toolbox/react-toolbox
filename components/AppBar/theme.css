@import '../colors.css';
@import '../media.css';
@import '../variables.css';
@import './config.css';

.appBar {
  background: var(--appbar-color);
  color: var(--appbar-contrast);
  font-family: var(--preferred-font);
  height: var(--appbar-height);
  padding: 0 var(--appbar-h-padding);
  transition-duration: 0.5s;
  transition-property: transform;
  transition-timing-function: cubic-bezier(0.215, 0.61, 0.355, 1);

  @apply --reset;

  &.scrollHide {
    transform: translateY(-100%);
  }

  &:not(.flat) {
    box-shadow: var(--shadow-4p);
  }

  &.fixed {
    left: 0;
    position: fixed;
    right: 0;
    top: 0;
    z-index: var(--z-index-highest);
  }

  & .inner {
    align-items: center;
    display: flex;
    height: 100%;
    position: relative;
    width: 100%;
  }

  & a {
    color: var(--appbar-contrast);
  }

  @media screen and (--xxs-viewport) and (--portrait) {
    height: var(--appbar-height-m-portrait);
  }

  @media screen and (--xs-viewport) and (--landscape) {
    height: var(--appbar-height-m-landscape);
  }
}

.title {
  flex-grow: 1;
  font-size: calc(1.8 * var(--unit));
  font-weight: bold;
  overflow-x: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;

  & > small {
    font-size: calc(1.8 * var(--unit));
    font-weight: normal;
  }
}

.leftIcon {
  margin-left: calc(-1.2 * var(--unit));
}

.rightIcon {
  margin-left: auto;
  margin-right: calc(-1.2 * var(--unit));
}
