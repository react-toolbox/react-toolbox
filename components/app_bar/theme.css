@import "../globals.css";
@import "./config.css";

.appBar {
  display: flex;
  height: var(--appbar-height);
  align-items: center;
  padding: 0 var(--appbar-h-padding);
  color: var(--appbar-contrast);
  background: var(--appbar-color);

  @media screen and (--layout-breakpoint-xxs) and (orientation: portrait) {
    height: var(--appbar-height-m-portrait);
  }

  @media screen and (--layout-breakpoint-xs) and (orientation: landscape) {
    height: var(--appbar-height-m-landscape);
  }

  &:not(.flat) {
    z-index: var(--z-index-high);
    box-shadow: 0 2px 5px rgba(0,0,0,.26);
  }

  &.fixed {
    position: fixed;
    top: 0;
    right: 0;
    left: 0;
    z-index: var(--z-index-highest);
  }

  & a {
    color: var(--appbar-contrast);
  }
}
