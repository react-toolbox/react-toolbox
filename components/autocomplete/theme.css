@import "../colors.css";
@import "../globals.css";
@import "./config.css";
@import "../input/config.css";

.autocomplete {
  position: relative;
  padding: var(--unit) 0;
  &.focus {
    & .label {
      color: var(--autocomplete-color-primary);
    }
    & .suggestions {
      max-height: var(--autocomplete-overflow-max-height);
      visibility: visible;
      box-shadow: var(--zdepth-shadow-1);
    }
  }
}

.label {
  font-size: var(--font-size-tiny);
  color: var(--color-text-secondary);
  transition: color var(--animation-duration) var(--animation-curve-default);
}

.values {
  flex-direction: row;
  flex-wrap: wrap;
  padding-bottom: calc(var(--unit) / 2);
}

.value {
  margin: var(--autocomplete-value-margin);
}

.suggestions {
  position: absolute;
  z-index: var(--z-index-high);
  width: 100%;
  max-height: 0;
  overflow-x: hidden;
  overflow-y: auto;
  visibility: hidden;
  background-color: var(--autocomplete-suggestions-background);
  transition-timing-function: var(--animation-curve-default);
  transition-duration: var(--animation-duration);
  transition-property: max-height, box-shadow;
  &:not(.up) {
    margin-top: calc(-1 * var(--input-padding));
  }
  &.up {
    bottom: 0;
  }
  /* no-webkit-scrollbar */
  &::-webkit-scrollbar {
    width: 0;
    height: 0;
  }
}

.suggestion {
  padding: var(--autocomplete-suggestion-padding);
  font-size: var(--input-field-font-size);
  cursor: pointer;
  &.active {
    background-color: var(--autocomplete-suggestion-active-background);
  }
}

.input {
  position: relative;
  &:after {
    position: absolute;
    top: 50%;
    right: var(--input-chevron-offset);
    width: 0;
    height: 0;
    pointer-events: none;
    content: "";
    border-top: calc(var(--input-field-height) / 7) solid var(--input-text-bottom-border-color);
    border-right: calc(var(--input-field-height) / 7) solid transparent;
    border-left: calc(var(--input-field-height) / 7) solid transparent;
    transition: transform var(--animation-duration) var(--animation-curve-default);
  }
}
