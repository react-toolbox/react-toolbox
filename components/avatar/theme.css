@import "../colors.css";
@import "../globals.css";
@import "./config.css";

.avatar {
  position: relative;
  display: inline-block;
  width: var(--avatar-size);
  height: var(--avatar-size);
  overflow: hidden;
  font-size: var(--avatar-font-size);
  color: var(--avatar-color);
  text-align: center;
  vertical-align: middle;
  background-color: var(--avatar-background);
  border-radius: 50%;
  & > svg {
    width: 1em;
    height: var(--avatar-size);
    fill: currentColor;
  }
  & > img {
    max-width: 100%;
    height: auto;
  }
}

.image {
  position: absolute;
  display: block;
  width: 100%;
  height: 100%;
  background-color: transparent;
  background-position: center;
  background-size: cover;
  border-radius: 50%;
}

.letter {
  display: block;
  width: 100%;
  line-height: var(--avatar-size);
}
