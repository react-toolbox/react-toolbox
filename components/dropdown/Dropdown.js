/* eslint-disable */
import React, { Component, PropTypes } from 'react';
import ReactDOM from 'react-dom';
import classnames from 'classnames';
import { themr } from 'react-css-themr';
import { DROPDOWN } from '../identifiers';
import InjectInput from '../input/Input';
import events from '../utils/events';

const factory = (Input) => {
  class Dropdown extends Component {
    static propTypes = {
      allowBlank: PropTypes.bool,
      auto: PropTypes.bool,
      className: PropTypes.string,
      disabled: PropTypes.bool,
      error: PropTypes.string,
      label: PropTypes.string,
      labelKey: PropTypes.string,
      name: PropTypes.string,
      onBlur: PropTypes.func,
      onChange: PropTypes.func,
      onClick: PropTypes.func,
      onFocus: PropTypes.func,
      required: PropTypes.bool,
      source: PropTypes.arrayOf(PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.object,
      ])).isRequired,
      template: PropTypes.func,
      theme: PropTypes.shape({
        active: PropTypes.string,
        disabled: PropTypes.string,
        dropdown: PropTypes.string,
        error: PropTypes.string,
        errored: PropTypes.string,
        field: PropTypes.string,
        label: PropTypes.string,
        required: PropTypes.string,
        selected: PropTypes.string,
        templateValue: PropTypes.string,
        up: PropTypes.string,
        value: PropTypes.string,
        values: PropTypes.string,
      }),
      value: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
      ]),
      valueKey: PropTypes.string,
    };

    static defaultProps = {
      auto: true,
      className: '',
      allowBlank: true,
      disabled: false,
      labelKey: 'label',
      required: false,
      valueKey: 'value',
    };

    state = {
      active: false,
      up: false,
    };

    componentWillUpdate(nextProps, nextState) {
      if (!this.state.active && nextState.active) {
        events.addEventsToDocument(this.getDocumentEvents());
      }
    }

    componentDidUpdate(prevProps, prevState) {
      if (prevState.active && !this.state.active) {
        events.removeEventsFromDocument(this.getDocumentEvents());
      }
    }

    componentWillUnmount() {
      if (this.state.active) {
        events.removeEventsFromDocument(this.getDocumentEvents());
      }
    }

    getDocumentEvents = () => ({
      click: this.handleDocumentClick,
      touchend: this.handleDocumentClick,
    });

    getSelectedItem = () => {
      for (const item of this.props.source) {
        if (item[this.props.valueKey] === this.props.value) return item;
      }
      return !this.props.allowBlank
        ? this.props.source[0]
        : undefined;
    };

    handleSelect = (item, event) => {
      if (this.props.onBlur) this.props.onBlur(event);
      if (!this.props.disabled && this.props.onChange) {
        if (this.props.name) event.target.name = this.props.name;
        this.props.onChange(item, event);
        this.close();
      }
    };

    handleClick = (event) => {
      this.open(event);
      events.pauseEvent(event);
      if (this.props.onClick) this.props.onClick(event);
    };

    handleDocumentClick = (event) => {
      if (this.state.active && !events.targetIsDescendant(event, ReactDOM.findDOMNode(this))) {
        this.setState({ active: false });
      }
    };

    close = () => {
      if (this.state.active) {
        this.setState({ active: false });
      }
    }

    open = (event) => {
      if (this.state.active) return;
      const client = event.target.getBoundingClientRect();
      const screenHeight = window.innerHeight || document.documentElement.offsetHeight;
      const up = this.props.auto ? client.top > ((screenHeight / 2) + client.height) : false;
      if (this.inputNode) this.inputNode.blur();
      this.setState({ active: true, up });
    };

    handleFocus = (event) => {
      event.stopPropagation();
      if (!this.props.disabled) this.open(event);
      if (this.props.onFocus) this.props.onFocus(event);
    };

    handleBlur = (event) => {
      event.stopPropagation();
      if (this.state.active) this.close();
      if (this.props.onBlur) this.props.onBlur(event);
    }

    renderTemplateValue(selected) {
      const { theme } = this.props;
      const className = classnames(theme.field, {
        [theme.errored]: this.props.error,
        [theme.disabled]: this.props.disabled,
        [theme.required]: this.props.required,
      });

      return (
        <div className={className} onClick={this.handleClick}>
          <div className={`${theme.templateValue} ${theme.value}`}>
            {this.props.template(selected)}
          </div>
          {this.props.label
            ? (
              <label className={theme.label}>
                {this.props.label}
                {this.props.required ? <span className={theme.required}> * </span> : null}
              </label>
            ) : null}
          {this.props.error ? <span className={theme.error}>{this.props.error}</span> : null}
        </div>
      );
    }

    renderValue = (item, idx) => {
      const { labelKey, theme, valueKey } = this.props;
      const className = classnames({
        [theme.selected]: item[valueKey] === this.props.value,
        [theme.disabled]: item.disabled,
      });
      return (
        <li
          key={idx}
          className={className}
          onClick={!item.disabled && this.handleSelect.bind(this, item[valueKey])}
        >
          {this.props.template ? this.props.template(item) : item[labelKey]}
        </li>
      );
    };

    render() {
      const {
        allowBlank, auto, labelKey, required, onChange, onFocus, onBlur, // eslint-disable-line no-unused-vars
        source, template, theme, valueKey, ...others
      } = this.props;
      const selected = this.getSelectedItem();
      const className = classnames(theme.dropdown, {
        [theme.up]: this.state.up,
        [theme.active]: this.state.active,
        [theme.disabled]: this.props.disabled,
        [theme.required]: this.props.required,
      }, this.props.className);

      return (
        <div
          className={className}
          data-react-toolbox="dropdown"
          onBlur={this.handleBlur}
          onFocus={this.handleFocus}
        >
          <Input
            {...others}
            tabIndex="0"
            className={theme.value}
            onClick={this.handleClick}
            required={this.props.required}
            readOnly
            ref={(node) => { this.inputNode = node && node.getWrappedInstance(); }}
            type={template && selected ? 'hidden' : null}
            theme={theme}
            themeNamespace="input"
            value={selected && selected[labelKey] ? selected[labelKey] : ''}
          />
          {template && selected ? this.renderTemplateValue(selected) : null}
          <ul className={theme.values}>
            {source.map(this.renderValue)}
          </ul>
        </div>
      );
    }
  }

  return Dropdown;
};

const Dropdown = factory(InjectInput);
export default themr(DROPDOWN)(Dropdown);
export { factory as dropdownFactory };
export { Dropdown };
