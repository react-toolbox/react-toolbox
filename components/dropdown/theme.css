@import '../colors.css';
@import '../variables.css';
@import '../input/config.css';
@import './config.css';

.dropdown {
  position: relative;

  @apply --reset;

  &:not(.active) {
    & > .values {
      max-height: 0;
      visibility: hidden;
    }
  }

  &.active {
    & > .label,
    & > .value {
      opacity: 0.5;
    }

    & > .values {
      box-shadow: var(--zdepth-shadow-1);
      max-height: var(--dropdown-overflow-max-height);
      visibility: visible;
    }
  }

  &:not(.up) > .values {
    bottom: auto;
    top: 0;
  }

  &.up > .values {
    bottom: 0;
    top: auto;
  }

  &.disabled {
    cursor: normal;
    pointer-events: none;
  }
}

.value {
  & > input {
    cursor: pointer;
  }

  &::after {
    border-left: var(--dropdown-value-border-size) solid transparent;
    border-right: var(--dropdown-value-border-size) solid transparent;
    border-top: var(--dropdown-value-border-size) solid var(--input-text-bottom-border-color);
    content: '';
    height: 0;
    pointer-events: none;
    position: absolute;
    right: var(--input-chevron-offset);
    top: 50%;
    transition: transform var(--animation-duration) var(--animation-curve-default);
    width: 0;
  }
}

.field {
  cursor: pointer;
  padding: var(--input-padding) 0;
  position: relative;

  &.errored {
    padding-bottom: 0;

    & > .label {
      color: var(--input-text-error-color);
    }

    & > .templateValue {
      border-bottom: 1px solid var(--input-text-error-color);
    }

    & > .label > .required {
      color: var(--input-text-error-color);
    }
  }

  &.disabled {
    cursor: normal;
    pointer-events: none;

    & > .templateValue {
      border-bottom-style: dotted;
      opacity: 0.7;
    }
  }
}

.templateValue {
  background-color: var(--input-text-background-color);
  border-bottom: 1px solid var(--input-text-bottom-border-color);
  color: var(--color-text);
  min-height: var(--input-field-height);
  padding: var(--input-field-padding) 0;
  position: relative;
}

.label {
  color: var(--input-text-label-color);
  font-size: var(--input-label-font-size);
  left: 0;
  line-height: var(--input-field-font-size);
  position: absolute;
  top: var(--input-focus-label-top);

  & .required {
    color: var(--input-text-error-color);
  }
}

.error {
  color: var(--input-text-error-color);
  font-size: var(--input-label-font-size);
  line-height: var(--input-underline-height);
  margin-bottom: calc(-1 * var(--input-underline-height));
}

.values {
  background-color: var(--dropdown-color-white);
  border-radius: var(--dropdown-value-border-radius);
  list-style: none;
  margin: 0;
  overflow-y: auto;
  padding: 0;
  position: absolute;
  transition-duration: var(--animation-duration);
  transition-property: max-height, box-shadow;
  transition-timing-function: var(--animation-curve-default);
  width: 100%;
  z-index: var(--z-index-high);

  & > * {
    cursor: pointer;
    overflow: hidden;
    padding: var(--unit);
    position: relative;

    &:hover:not(.disabled),
    &.focused:not(.disabled) {
      background-color: var(--dropdown-value-hover-background);
      outline: none;
    }

    &.selected {
      color: var(--dropdown-color-primary);
    }

    &.disabled {
      color: var(--dropdown-color-disabled);
      cursor: not-allowed;
    }
  }

  &::-webkit-scrollbar {
    height: 0;
    width: 0;
  }
}
