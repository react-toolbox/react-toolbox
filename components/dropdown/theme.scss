@import "../colors";
@import "../globals";
@import "../mixins";
@import "../input/config";
@import "./config";

.dropdown {
  position: relative;
  &:not(.active) {
    > .values {
      max-height: 0;
      visibility: hidden;
    }
  }
  &.active {
    > .label, > .value {
      opacity: .5;
    }
    > .values {
      max-height: $dropdown-overflow-max-height;
      visibility: visible;
      box-shadow: $zdepth-shadow-1;
    }
  }
  &:not(.up) > .values {
    top: 0;
    bottom: auto;
  }
  &.up > .values {
    top: auto;
    bottom: 0;
  }
  &.disabled {
    pointer-events: none;
    cursor: normal;
  }
  &.menu {
    &.segmented {
      .value {
        display: flex;

        &:after {
          display: none;
        }

        span {
          display: none;
        }
        input, .segmentedButton {
          &:hover {
            background-color: $dropdown-input-hover-background;
          }
        }
      }
      .values {
        .menuValue {
          display: flex;
          justify-content: space-between;

          .segmentedButton {
            flex: 0 0 4.45rem;
            margin: -$dropdown-menu-input-padding;
            border-top: 0;
            border-bottom: 0;

            &:after {
              @include arrow-up;
            }
          }
          &:after {
            display: none;
          }
        }
      }
      .segmentedButton {
        width: 4.5rem;
        cursor: pointer;
        background-color: transparent;
        border: 0;
        border-top: 1px solid $input-text-bottom-border-color;
        border-bottom: 1px solid $input-text-bottom-border-color;
        border-left: 1px solid $input-text-bottom-border-color;
        appearance: none;

      }
      &:not(.active) .segmentedButton {
        &:after {
          @include arrow-down;
        }
      }
    }
    input {
      padding: $dropdown-menu-input-padding 1.6rem;
      border-top: 1px solid $input-text-bottom-border-color;

      &:hover {
        background-color: $dropdown-input-hover-background;
      }
    }
    .value {
      label {
        top: 3.6rem;
        padding-left: $dropdown-menu-input-padding;
      }
      &:after {
        right: 1.6rem;
      }
    }
    .values {
      top: 2rem;
    }
    .menuValue {
      padding: 1.5rem;
      border-bottom: 1px solid $input-text-bottom-border-color;
      &:after {
        @include arrow-up;
      }
    }
    &.up {
      > .values {
        top: auto;
        bottom: 2rem;
      }
      .menuValue {
        border-top: 1px solid $input-text-bottom-border-color;
        border-bottom: 0;
      }
    }
  }
}

.value {
  > input {
    cursor: pointer;
  }
  &:after {
    $size: ($input-field-height / 7);
    $border: $size solid transparent;
    position: absolute;
    top: 50%;
    right: $input-chevron-offset;
    width: 0;
    height: 0;
    pointer-events: none;
    content: "";
    border-top: $size solid $input-text-bottom-border-color;
    border-right: $border;
    border-left: $border;
    transition: transform $animation-duration $animation-curve-default;
  }
}

.field {
  position: relative;
  padding: $input-padding 0;
  cursor: pointer;
  &.errored {
    padding-bottom: 0;
    > .label {
      color: $input-text-error-color;
    }
    > .templateValue {
      border-bottom: 1px solid $input-text-error-color;
    }
    > .label > .required {
      color: $input-text-required-color;
    }
  }
  &.disabled {
    pointer-events: none;
    cursor: normal;
    > .templateValue {
      border-bottom-style: dotted;
      opacity: .7;
    }
  }
}

.templateValue {
  position: relative;
  min-height: $input-field-height;
  padding: $input-field-padding 0;
  color: $color-text;
  background-color: $input-text-background-color;
  border-bottom: 1px solid $input-text-bottom-border-color;
}

.label {
  position: absolute;
  top: $input-focus-label-top;
  left: 0;
  font-size: $input-label-font-size;
  line-height: $input-field-font-size;
  color: $input-text-label-color;
  .required {
    color: $input-text-required-color;
  }
}

.error {
  margin-bottom: - $input-underline-height;
  font-size: $input-label-font-size;
  line-height: $input-underline-height;
  color: $input-text-error-color;
}

.values {
  @include no-webkit-scrollbar;
  position: absolute;
  z-index: $z-index-high;
  width: 100%;
  overflow-y: auto;
  list-style: none;
  background-color: $dropdown-color-white;
  border-radius: $dropdown-value-border-radius;
  transition-timing-function: $animation-curve-default;
  transition-duration: $animation-duration;
  transition-property: max-height, box-shadow;
  > * {
    position: relative;
    padding: $unit;
    overflow: hidden;
    cursor: pointer;
    &:hover {
      background-color: $dropdown-value-hover-background;
    }
    &.selected {
      color: $dropdown-color-primary;
    }
  }
}
