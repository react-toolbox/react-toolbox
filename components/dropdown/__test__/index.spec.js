import expect from 'expect';
import React from 'react';
import {
  renderIntoDocument,
  scryRenderedDOMComponentsWithClass,
  Simulate
} from 'react-addons-test-utils';
import sinon from 'sinon';
import theme from '../theme.css';
import Dropdown from '../Dropdown';


describe('Dropdown', function () {
  describe('#renderValue', function () {
    const source = [
      { value: 'EN-gb', label: 'England' },
      { value: 'ES-es', label: 'Spain', disabled: true },
      { value: 'TH-th', label: 'Thailand', disabled: true },
      { value: 'EN-en', label: 'USA'}
    ];
    it('renders dropdown item with disabled style', function () {
      const tree = renderIntoDocument(<Dropdown theme={theme} source={source} />);
      const disabled = scryRenderedDOMComponentsWithClass(tree, theme.disabled);
      expect(disabled.length).toEqual(2);
    });
    it('does not call onChange callback when disabled dorpdown item is clicked', function () {
      const spy = sinon.spy();
      const tree = renderIntoDocument(<Dropdown
        theme={theme}
        source={source}
        value={source[0].value}
        onChange={spy}
      />);
      const disabled = scryRenderedDOMComponentsWithClass(tree, theme.disabled);
      expect(spy.called).toEqual(false);
      Simulate.click(disabled[0]);
      expect(spy.called).toEqual(false);
      const selected = scryRenderedDOMComponentsWithClass(tree, theme.selected);
      Simulate.click(selected[0]);
      expect(spy.called).toEqual(true);
    });
  });
});
