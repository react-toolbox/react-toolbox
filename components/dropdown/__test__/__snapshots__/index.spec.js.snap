exports[`Dropdown snapshot testing Spain selected 1`] = `
<div
  className="dropdown"
  data-react-toolbox="dropdown"
  onBlur={[Function]}
  onFocus={[Function]}>
  <Input
    className="value"
    disabled={false}
    floating={true}
    hint=""
    multiline={false}
    onClick={[Function]}
    readOnly={true}
    required={false}
    tabIndex="0"
    theme={Object {}}
    themeNamespace="input"
    type={null}
    value="Spain" />
  <ul
    className="values">
    <li
      className=""
      onClick={[Function]}>
      England
    </li>
    <li
      className="selected"
      onClick={[Function]}>
      Spain
    </li>
    <li
      className="disabled"
      onClick={false}>
      Thailand
    </li>
    <li
      className=""
      onClick={[Function]}>
      USA
    </li>
  </ul>
</div>
`;

exports[`Dropdown snapshot testing allowBlank=false 1`] = `
<div
  className="dropdown"
  data-react-toolbox="dropdown"
  onBlur={[Function]}
  onFocus={[Function]}>
  <Input
    className="value"
    disabled={false}
    floating={true}
    hint=""
    multiline={false}
    onClick={[Function]}
    readOnly={true}
    required={false}
    tabIndex="0"
    theme={Object {}}
    themeNamespace="input"
    type={null}
    value="England" />
  <ul
    className="values">
    <li
      className=""
      onClick={[Function]}>
      England
    </li>
    <li
      className=""
      onClick={[Function]}>
      Spain
    </li>
    <li
      className="disabled"
      onClick={false}>
      Thailand
    </li>
    <li
      className=""
      onClick={[Function]}>
      USA
    </li>
  </ul>
</div>
`;

exports[`Dropdown snapshot testing disabled 1`] = `
<div
  className="dropdown disabled"
  data-react-toolbox="dropdown"
  onBlur={[Function]}
  onFocus={[Function]}>
  <Input
    className="value"
    disabled={true}
    floating={true}
    hint=""
    multiline={false}
    onClick={[Function]}
    readOnly={true}
    required={false}
    tabIndex="0"
    theme={Object {}}
    themeNamespace="input"
    type={null}
    value="" />
  <ul
    className="values">
    <li
      className=""
      onClick={[Function]}>
      England
    </li>
    <li
      className=""
      onClick={[Function]}>
      Spain
    </li>
    <li
      className="disabled"
      onClick={false}>
      Thailand
    </li>
    <li
      className=""
      onClick={[Function]}>
      USA
    </li>
  </ul>
</div>
`;

exports[`Dropdown snapshot testing none selected 1`] = `
<div
  className="dropdown"
  data-react-toolbox="dropdown"
  onBlur={[Function]}
  onFocus={[Function]}>
  <Input
    className="value"
    disabled={false}
    floating={true}
    hint=""
    multiline={false}
    onClick={[Function]}
    readOnly={true}
    required={false}
    tabIndex="0"
    theme={Object {}}
    themeNamespace="input"
    type={null}
    value="" />
  <ul
    className="values">
    <li
      className=""
      onClick={[Function]}>
      England
    </li>
    <li
      className=""
      onClick={[Function]}>
      Spain
    </li>
    <li
      className="disabled"
      onClick={false}>
      Thailand
    </li>
    <li
      className=""
      onClick={[Function]}>
      USA
    </li>
  </ul>
</div>
`;

exports[`Dropdown snapshot testing required=true 1`] = `
<div
  className="dropdown required"
  data-react-toolbox="dropdown"
  onBlur={[Function]}
  onFocus={[Function]}>
  <Input
    className="value"
    disabled={false}
    floating={true}
    hint=""
    multiline={false}
    onClick={[Function]}
    readOnly={true}
    required={true}
    tabIndex="0"
    theme={Object {}}
    themeNamespace="input"
    type={null}
    value="" />
  <ul
    className="values">
    <li
      className=""
      onClick={[Function]}>
      England
    </li>
    <li
      className=""
      onClick={[Function]}>
      Spain
    </li>
    <li
      className="disabled"
      onClick={false}>
      Thailand
    </li>
    <li
      className=""
      onClick={[Function]}>
      USA
    </li>
  </ul>
</div>
`;

exports[`Dropdown snapshot testing with className 1`] = `
<div
  className="dropdown something fancy"
  data-react-toolbox="dropdown"
  onBlur={[Function]}
  onFocus={[Function]}>
  <Input
    className="value"
    disabled={false}
    floating={true}
    hint=""
    multiline={false}
    onClick={[Function]}
    readOnly={true}
    required={false}
    tabIndex="0"
    theme={Object {}}
    themeNamespace="input"
    type={null}
    value="" />
  <ul
    className="values">
    <li
      className=""
      onClick={[Function]}>
      England
    </li>
    <li
      className=""
      onClick={[Function]}>
      Spain
    </li>
    <li
      className="disabled"
      onClick={false}>
      Thailand
    </li>
    <li
      className=""
      onClick={[Function]}>
      USA
    </li>
  </ul>
</div>
`;

exports[`Dropdown snapshot testing with error message 1`] = `
<div
  className="dropdown"
  data-react-toolbox="dropdown"
  onBlur={[Function]}
  onFocus={[Function]}>
  <Input
    className="value"
    disabled={false}
    error="some error message"
    floating={true}
    hint=""
    multiline={false}
    onClick={[Function]}
    readOnly={true}
    required={false}
    tabIndex="0"
    theme={Object {}}
    themeNamespace="input"
    type={null}
    value="" />
  <ul
    className="values">
    <li
      className=""
      onClick={[Function]}>
      England
    </li>
    <li
      className=""
      onClick={[Function]}>
      Spain
    </li>
    <li
      className="disabled"
      onClick={false}>
      Thailand
    </li>
    <li
      className=""
      onClick={[Function]}>
      USA
    </li>
  </ul>
</div>
`;

exports[`Dropdown snapshot testing with label 1`] = `
<div
  className="dropdown"
  data-react-toolbox="dropdown"
  onBlur={[Function]}
  onFocus={[Function]}>
  <Input
    className="value"
    disabled={false}
    floating={true}
    hint=""
    label="Select country"
    multiline={false}
    onClick={[Function]}
    readOnly={true}
    required={false}
    tabIndex="0"
    theme={Object {}}
    themeNamespace="input"
    type={null}
    value="" />
  <ul
    className="values">
    <li
      className=""
      onClick={[Function]}>
      England
    </li>
    <li
      className=""
      onClick={[Function]}>
      Spain
    </li>
    <li
      className="disabled"
      onClick={false}>
      Thailand
    </li>
    <li
      className=""
      onClick={[Function]}>
      USA
    </li>
  </ul>
</div>
`;

exports[`Dropdown snapshot testing with minimum properties 1`] = `
<div
  className="dropdown"
  data-react-toolbox="dropdown"
  onBlur={[Function]}
  onFocus={[Function]}>
  <Input
    className="value"
    disabled={false}
    floating={true}
    hint=""
    multiline={false}
    onClick={[Function]}
    readOnly={true}
    required={false}
    tabIndex="0"
    theme={Object {}}
    themeNamespace="input"
    type={null}
    value="" />
  <ul
    className="values">
    <li
      className=""
      onClick={[Function]}>
      England
    </li>
    <li
      className=""
      onClick={[Function]}>
      Spain
    </li>
    <li
      className="disabled"
      onClick={false}>
      Thailand
    </li>
    <li
      className=""
      onClick={[Function]}>
      USA
    </li>
  </ul>
</div>
`;

exports[`Dropdown snapshot testing with template 1`] = `
<div
  className="dropdown"
  data-react-toolbox="dropdown"
  onBlur={[Function]}
  onFocus={[Function]}>
  <Input
    className="value"
    disabled={false}
    floating={true}
    hint=""
    multiline={false}
    onClick={[Function]}
    readOnly={true}
    required={false}
    tabIndex="0"
    theme={Object {}}
    themeNamespace="input"
    type={null}
    value="" />
  <ul
    className="values">
    <li
      className=""
      onClick={[Function]}>
      {\"value\":\"EN-gb\",\"label\":\"England\"}
    </li>
    <li
      className=""
      onClick={[Function]}>
      {\"value\":\"ES-es\",\"label\":\"Spain\"}
    </li>
    <li
      className="disabled"
      onClick={false}>
      {\"value\":\"TH-th\",\"label\":\"Thailand\",\"disabled\":true}
    </li>
    <li
      className=""
      onClick={[Function]}>
      {\"value\":\"EN-en\",\"label\":\"USA\"}
    </li>
  </ul>
</div>
`;
